src/DateField.tsx

```diff
@@ -5,7 +5,15 @@ import { connectField } from 'uniforms/es5';
 import wrapField from './wrapField';
 
 const DateConstructor = (typeof global === 'object' ? global : window).Date;
-const dateFormat = value => value && value.toISOString();
+const dateFormat = (value) => {
+  if(typeof value === "string"){
+    return value;
+  } else if(value.toISOString){
+    value && value.toISOString();
+  } else {
+    return undefined;
```

Comment:
@kingsleyzissou Not sure how to handle this case. 
Data is invalid. Logging warning can be excessive here but if object is not string or date we cannot do much